File: backend/app/schemas/item.py
from datetime import datetime, date
from typing import Optional
from pydantic import BaseModel


class ItemBase(BaseModel):
    name: str
    quantity: int = 1
    price: float = 0.0
    purchase_date: Optional[date] = None
    expiry_date: Optional[date] = None
    category_id: Optional[int] = None
    room_id: Optional[int] = None
    family_id: Optional[int] = None  # 所属家庭
    is_private: bool = False  # 仅自己可见
    location: Optional[str] = None
    notes: Optional[str] = None
    usage: Optional[str] = None
    purchase_channel: Optional[str] = None


class ItemCreate(ItemBase):
    pass


class ItemUpdate(BaseModel):
    name: Optional[str] = None
    quantity: Optional[int] = None
    price: Optional[float] = None
    purchase_date: Optional[date] = None
    expiry_date: Optional[date] = None
    category_id: Optional[int] = None
    room_id: Optional[int] = None
    family_id: Optional[int] = None
    is_private: Optional[bool] = None
    location: Optional[str] = None
    notes: Optional[str] = None
    usage: Optional[str] = None
    purchase_channel: Optional[str] = None


class ItemResponse(ItemBase):
    id: int
    user_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ItemListResponse(BaseModel):
    items: list[ItemResponse]
    total: int

File: frontend/src/api/item.ts
import api from './index'
import type { Item, ItemListResponse } from '@/types'

export interface ItemCreate {
  name: string
  quantity?: number
  price?: number
  purchase_date?: string | null
  expiry_date?: string | null
  category_id?: number | null
  location?: string | null
  notes?: string | null
  usage?: string | null
  purchase_channel?: string | null
  room_id?: number | null
}

export interface ItemUpdate {
  name?: string
  quantity?: number
  price?: number
  purchase_date?: string | null
  expiry_date?: string | null
  category_id?: number | null
  location?: string | null
  notes?: string | null
  usage?: string | null
  purchase_channel?: string | null
  room_id?: number | null
}

export interface ItemQueryParams {
  name?: string
  category_id?: number
  expired?: boolean
  expiring_soon?: boolean
  skip?: number
  limit?: number
  room_id?: number
}

export const itemApi = {
  getAll: async (params?: ItemQueryParams): Promise<ItemListResponse> => {
    const response = await api.get<ItemListResponse>('/items', { params })
    return response.data
  },

  getById: async (id: number): Promise<Item> => {
    const response = await api.get<Item>(`/items/${id}`)
    return response.data
  },

  create: async (data: ItemCreate): Promise<Item> => {
    const response = await api.post<Item>('/items', data)
    return response.data
  },

  update: async (id: number, data: ItemUpdate): Promise<Item> => {
    const response = await api.put<Item>(`/items/${id}`, data)
    return response.data
  },

  delete: async (id: number): Promise<void> => {
    await api.delete(`/items/${id}`)
  }
}
